Fix test events filter container

DarianAnjuhal opened the pull request:
Fixed pretty simple issue:
https://github.com/docker/docker/issues/12660

I needed time to understand whats going on with compiling, testing and the go language (which is weird).

tianon committed:
Fix daemon start/stop logic in hack/make/* scripts

From the Bash manual's `set -e` description:
(https://www.gnu.org/software/bash/manual/bashref.html#index-set)

> Exit immediately if a pipeline (see Pipelines), which may consist of a
> single simple command (see Simple Commands), a list (see Lists), or a
> compound command (see Compound Commands) returns a non-zero status.
> The shell does not exit if the command that fails is part of the
> command list immediately following a while or until keyword, part of
> the test in an if statement, part of any command executed in a && or
> || list except the command following the final && or ||, any command
> in a pipeline but the last, or if the command’s return status is being
> inverted with !. If a compound command other than a subshell returns a
> non-zero status because a command failed while -e was being ignored,
> the shell does not exit.

Additionally, further down:

> If a compound command or shell function executes in a context where -e
> is being ignored, none of the commands executed within the compound
> command or function body will be affected by the -e setting, even if
> -e is set and a command returns a failure status. If a compound
> command or shell function sets -e while executing in a context where
> -e is ignored, that setting will not have any effect until the
> compound command or the command containing the function call
> completes.

Thus, the only way to have our `.integration-daemon-stop` script
actually run appropriately to clean up our daemon on test/script failure
is to use `trap ... EXIT`, which we traditionally avoid because it does
not have any stacking capabilities, but in this case is a reasonable
compromise because it's going to be the only script using it (for now,
at least; we can evaluate more complex solutions in the future if they
actually become necessary).

The alternatives were much less reasonable.  One is to have the entire
complex chains in any script wanting to use `.integration-daemon-start`
/ `.integration-daemon-stop` be chained together with `&&` in an `if`
block, which is untenable.  The other I could think of was taking the
body of these scripts out into separate scripts, essentially meaning
we'd need two files for each of these, which further complicates the
maintenance.

Add to that the fact that our `trap ... EXIT` is scoped to the enclosing
subshell (`( ... )`) and we're in even more reasonable territory with
this pattern.

Signed-off-by: Andrew "Tianon" Page <admwiggin@gmail.com>

jessfraz committed:
actually depreciate -rm, -sig-proxy, -name, seriously its been forever

Signed-off-by: Jessica Frazelle <princess@docker.com>

rajdeepd committed:
TestCase added for Container Create with HostName

Signed-off-by: Rajdeep Dua <dua_rajdeep@yahoo.com>

Commit:
Move CLI config processing out from under registry dir

No logic changes should be in here, just moving things around.

Signed-off-by: Doug Davis <dug@us.ibm.com>

Commit:
Validate we're not using the old testing stuff

Signed-off-by: Doug Davis <dug@us.ibm.com>

GordonTheTurtle commented:
Can you please sign your commits following these rules:
https://github.com/docker/docker/blob/master/CONTRIBUTING.md#sign-your-work
The easiest way to do this is to amend the last commit:
~~~console
$ git clone -b "fix-test-events-filter-container" git@github.com:DarianAnjuhal/docker.git somewhere
$ cd somewhere
$ git rebase -i HEAD~3
editor opens
change each 'pick' to 'edit'
save the file and quit
$ git commit --amend -s --no-edit
$ git rebase --continue # and repeat the amend for each commit
$ git push -f
~~~
This will update the existing PR, so you do not need to open a new one.

DarianAnjuhal commented:
I did the signature stuff already, should it automatically update or did I an error?

duglin commented:
You need to sign each commit, so squashing it down to one will fix it.

LK4D4 commented:
Lol, thanks. Also you need to check error from wait inspect.

icecrime commented on integration-cli/docker_cli_events_test.go:
```diff
@@ -327,7 +327,7 @@ func (s *DockerSuite) TestEventsFilterContainer(c *check.C) {
 		}
 		nameID[name] = strings.TrimSpace(out)
-		waitInspect(name, "{{.State.Runing }}", "false", 5)
+		waitInspect(name, "{{.State.Running }}", "false", 5)
```
Can we please replace this by a call to [`waitRun`](https://github.com/docker/docker/blob/eae272f90eb6c01824112054140b5dc7d9d679ce/integration-cli/utils.go#L202-L204)?

DarianAnjuhal replied:
Okay, I will change it.

DarianAnjuhal commented:
He he, I knew this would be to easy. :-)

icecrime commented:
@DarianAnjuhal It's not that easy, but don't worry it shouldn't be too hard ;-)
* [Squash your commits](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html)
* Replace the call to `waitInspect` by [`waitRun`](https://github.com/docker/docker/blob/eae272f90eb6c01824112054140b5dc7d9d679ce/integration-cli/utils.go#L202-L204)
* Verify `waitRun` return value as suggested by @LK4D4 
* Profit! (well, not really, this is open source, but thanks!)

WeiZhang555 committed:
refactor httpError() and add 404 "not found" mapping

When docker pull a non-existent repo, daemon will report "image xxx not found"
with an error code 500, which should be 404.
This commit add 404 "not found" mapping and refactor httpError function.

Signed-off-by: Zhang Wei <zhangwei555@huawei.com>

cpuguy83 committed:
Fix race with TestContainerApiCommit

Signed-off-by: Brian Goff <cpuguy83@gmail.com>

Commit:
Fix TestRenameStoppedContainer race

Signed-off-by: Antonio Murdaca <me@runcom.ninja>

LK4D4 committed:
Return weird behaviour of returning json errors

We need this, so client can get error from stream and not from status
code, which is already 200, because write to ResponseWriter was occured.

Signed-off-by: Alexander Morozov <lk4d4@docker.com>

GordonTheTurtle commented:
Looks like we would not be able to merge this PR because of merge conflicts. Please rebase, fix conflicts, and force push to your branch.

LK4D4 committed:
Merge pull request #12500 from tianon/fix-daemon-logic

Fix daemon start/stop logic in hack/make/* scripts

Commit:
Merge pull request #12687 from WeiZhang555/NotFound

refactor httpError() and add 404 "not found" mapping

LK4D4 committed:
Merge pull request #12646 from duglin/validateTests

Validate we're not using the old testing stuff

cpuguy83 committed:
Merge pull request #12703 from LK4D4/carry_12596

remove job from image_export

Commit:
Merge pull request #12692 from cpuguy83/fix_commit_test

Fix race with TestContainerApiCommit

Commit:
Merge pull request #12690 from rajdeepd/dry-run-test

Container API Test with HostName

DarianAnjuhal committed:
fix runtime issue for TestEventsFilterContainer

Signed-off-by: Daniel Antlinger <d.antlinger@gmx.at>

LK4D4 committed:
Merge pull request #12636 from duglin/MoveConfig

Move CLI config processing out from under registry dir

Commit:
Merge pull request #12577 from jfrazelle/actually-depreciate-rm

actually depreciate -rm insteam of --rm.. and a few others

Commit:
Merge pull request #12702 from LK4D4/good_ol_flushed

Return weird behaviour of returning json errors

Commit:
Merge pull request #12693 from runcom/fix-TestRenameStoppedContainer-race

Fix TestRenameStoppedContainer race

GordonTheTurtle commented:
Can you please sign your commits following these rules:
https://github.com/docker/docker/blob/master/CONTRIBUTING.md#sign-your-work
The easiest way to do this is to amend the last commit:
~~~console
$ git clone -b "fix-test-events-filter-container" git@github.com:DarianAnjuhal/docker.git somewhere
$ cd somewhere
$ git rebase -i HEAD~20
editor opens
change each 'pick' to 'edit'
save the file and quit
$ git commit --amend -s --no-edit
$ git rebase --continue # and repeat the amend for each commit
$ git push -f
~~~
This will update the existing PR, so you do not need to open a new one.

DarianAnjuhal commented:
Weired, there is the Signed-off-by: ... stuff in the comment and it still doesnt work?
